File: CS/ViewModel.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace MasterDetailInside {
    public class ViewModel {
        public ViewModel() {
            ParentData = new DataTable();
            ParentData.Columns.Add("Id", typeof(int));
            ParentData.Columns.Add("Text", typeof(string));
            ChildData = new DataTable();
            ChildData.Columns.Add("ParentId", typeof(int));
            ChildData.Columns.Add("Text", typeof(string));
            for (int i = 0; i < 100; i++) {
                ParentData.Rows.Add(i, "Master" + i);
                for (int j = 0; j < 50; j++)
                    ChildData.Rows.Add(i, "Detail" + j + " Master" + i);
            }
        }
        public DataTable ParentData { get; set; }
        public DataTable ChildData { get; set; }
    }
}

File: VB/ViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Text

Namespace MasterDetailInside
    Public Class ViewModel
        Public Sub New()
            ParentData = New DataTable()
            ParentData.Columns.Add("Id", GetType(Integer))
            ParentData.Columns.Add("Text", GetType(String))
            ChildData = New DataTable()
            ChildData.Columns.Add("ParentId", GetType(Integer))
            ChildData.Columns.Add("Text", GetType(String))
            For i As Integer = 0 To 99
                ParentData.Rows.Add(i, "Master" & i)
                For j As Integer = 0 To 49
                    ChildData.Rows.Add(i, "Detail" & j & " Master" & i)
                Next j
            Next i
        End Sub
        Public Property ParentData() As DataTable
        Public Property ChildData() As DataTable
    End Class
End Namespace
